kernel/mmu.c

```diff
@@ -555,6 +555,11 @@ static int map_anon_page(void *addr, uint32_t flags)
 		k_mem_page_frame_set(pf, K_MEM_PAGE_FRAME_PINNED);
 	}
 	frame_mapped_set(pf, addr);
+#ifdef CONFIG_DEMAND_PAGING
```

Comment:
Style: would be cleaner IMHO to not gate the k_mem_paging_eviction_*() APIs behind a kconfig, and just stub them as empty inline defaults, or provided by a per-algorithm header (where they could be empty for NRU), etc.. Mixed preprocessor and logic gets out of hand quickly.